src/fetch/block_number.py

```diff
@@ -0,0 +1,42 @@
+"""
+Fetches the closest block number for a given timestamp from Etherscan API
```

Comment:
Why have you chosen to add a new dependency? Wouldn't we also be able to get the information from dune?
E.g.
```
select * from ethereum.blocks limit 10
```
seems to contain all the information that you need

Reply:
Etherscan is much faster and free to fetch this information. Dune requires dependency on a legacy client that we should move away from. For the new dune client - do we really want to spend money per execution for such a basic piece of information that can be fetched for free from elsewhere?

Reply:
I think that we should limit our dune dependencies to those queries that we actually need it for and not elementary stuff that can be retrieved easily elsewhere.

Reply:
If we are buying an enterprise license anyways, I would not expect a significant cost. Maybe only some cents per execution. 
To me that is worth not having to deal with many changing external dependencies. But I understand the trade-off.

Reply:
I would be willing to turn this into a dune query if you think it important to limit the number of sources, however there are a few points that are worth consideration:

1. I think the SQL query is a bit bulkier and less concise than this very practical etherscan endpoint.
2. We can be almost certain that this endpoint will return results faster.
3. Even with enterprise, dune has a limit to the number of parallel executions (if we ever to make this async then these two block number requests would add to our parallel execution limit). 
4. This also serves as an example of how our data infrastructure can handle multiple data sources. Given that we are already introducing a dependency on the orderbook DB, it seems like a good time to introduce this additional data feed.

In general, I believe that etherscan offers a pretty nice API interface that we should consider using whenever possible (also their pricing plans are much cheaper than dune).

Would you like me to implement the get_block method using dune to demonstrate how much bulkier it would look?

Reply:
I don't have a super strong opinion. I just wanna avoid the situation where we need to manually look into things because another dependency is down.

- At least your implementation should have some error handling and automatic retry etc in order to avoid this scenario

Reply:
I have completely removed all traces of using the Etherscan API in favour of the Dune approach (which actually turned out much easier and clean).

Check this out: https://dune.com/queries/1328116